lighthouse-core/test/audits/dobetterweb/charset-test.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license Copyright 2020 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const CharsetDefinedAudit = require('../../../audits/dobetterweb/charset.js');
+const assert = require('assert');
+const networkRecordsToDevtoolsLog = require('../../network-records-to-devtools-log.js');
+
+/* eslint-env jest */
+
+function generateArtifacts(htmlContent, contentTypeValue = 'text/html') {
+  const finalUrl = 'https://example.com/';
+  const mainResource = {
+    url: finalUrl,
+    responseHeaders: [
+      {name: 'content-type', value: contentTypeValue},
+    ],
+  };
+  const devtoolsLog = networkRecordsToDevtoolsLog([mainResource]);
+  return {
+    devtoolsLogs: {[CharsetDefinedAudit.DEFAULT_PASS]: devtoolsLog},
+    URL: {finalUrl},
+    MainDocumentContent: htmlContent,
+  };
+}
+
+describe('Charset defined audit', () => {
+  it('succeeds when the page contains the charset meta tag', () => {
+    const htmlContent = '<meta charset="utf-8" />';
```

Comment:
to make these htmlContent stubs a _little_ more realistic i'd recommend a doctype and some content after the meta... like

`<!doctype html><meta charset="utf-8" /><h1>hello`